src/index.js

```diff
@@ -1,41 +1,84 @@
-    index: 4,
-  },
-  {
-    title: 'work on Personal soft skilss',
-    completed: false,
-    index: 5,
-  },
-];
+const newTodoForm = document.querySelector('#form-sec');
+const todoList = document.querySelector('.task-section');
+
+const updateInputText = (id, newText) => {
+  const todoListArray = JSON.parse(localStorage.getItem('todos') || '[]');
+  const targetIndex = todoListArray.findIndex((todo) => todo.id === parseInt(id, 10));
+
+  if (targetIndex >= 0) {
+    todoListArray[targetIndex].description = newText;
+    localStorage.setItem('todos', JSON.stringify(todoListArray));
+    tasksRefactor(todoListArray);
+  }
+};
+
+const toggleComplete = (id) => {
+  const todoListArray = JSON.parse(localStorage.getItem('todos') || '[]');
+  const updateTodoList = todoListArray.map((todo) => {
+    if (todo.id === parseInt(id, 10)) {
+      return { ...todo, completed: !todo.completed };
+    }
+    return todo;
+  });
+
+  tasksRefactor(updateTodoList);
+  // eslint-disable-next-line no-restricted-globals
+  location.reload();
+};
+
+const removeTodo = (targetIndex) => {
+  const todoListArr = JSON.parse(localStorage.getItem('todos') || '[]');
+  const updateList = todoListArr.filter((todo) => todo.id !== parseInt(targetIndex, 10));
+  tasksRefactor(updateList);
+  // eslint-disable-next-line no-restricted-globals
+  location.reload();
+};
 
-const description = todos.map((todo) => `
+const updateList = (todos) => {
+  const description = todos.map((todo) => `
             <li class="fill task-list ${todo.completed ? 'completed' : ''}" data-id="${todo.id}">
               <input type="checkbox" ${todo.completed ? 'checked' : ''} class="checkbox"/>
-              <input type="text" value="${todo.title}" class="inputtext" id="${todo.id}"/>
+              <input type="text" value="${todo.description}" class="inputtext" id="${todo.id}"/>
               <button type="button"><i class="fa-solid fa-trash-can"></i></button>
             </li>
           `).join('');
 
-const listContainer = document.querySelector('.list');
+  todoList.innerHTML = description;
+  const enteredTexts = todoList.querySelectorAll('.task-list input[type=text]');
+
+  enteredTexts.forEach((input) => input.addEventListener('change', (e) => updateInputText(input.id, e.target.value)));
+
+  const erase = todoList.querySelectorAll('.task-list button');
+
+  erase.forEach((button) => button.addEventListener('click', () => removeTodo(button.parentNode.getAttribute('data-id'))));
+
+  const checkedUP = todoList.querySelectorAll('.checkbox');
+  checkedUP.forEach((checkbox) => checkbox.addEventListener('click', () => toggleComplete(checkbox.parentNode.getAttribute('data-id'))));
+};
+
+const newTodo = (e) => {
+  e.preventDefault();
+  const newTask = document.getElementById('taskInput').value;
+  const todos = JSON.parse(localStorage.getItem('todos') || '[]');
+
+  const newTodo = {
+    description: newTask,
+    completed: false,
+    id: Date.now(),
+  };
+
```

Comment:
The index property should be set to the value of the new array length as mentioned in the requirments, currently you have set it as a Date. Kindly fix this issue.


![image](https://user-images.githubusercontent.com/87448628/236511911-7ae48134-cb4e-4edb-adf9-11555fec60f2.png)

Reply:
@Danieldotcomcoder 👋 thank you for pointing that out, I have now fixed it